rpc/types/src/bin.rs

```diff
@@ -134,6 +115,285 @@ define_request_and_response! {
+    daemon_time: u64,
+    added_pool_txs: Vec<PoolTxInfo>,
+    remaining_added_pool_txids: ByteArrayVec<32>,
+    removed_pool_txids: ByteArrayVec<32>,
+}
+
+/// Data within [`GetBlocksResponse::PoolInfoFull`].
+#[allow(dead_code, missing_docs)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+#[derive(Clone, Default, Debug, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct GetBlocksResponsePoolInfoFull {
+    pub status: Status,
+    pub untrusted: bool,
+    pub blocks: Vec<BlockCompleteEntry>,
+    pub start_height: u64,
+    pub current_height: u64,
+    pub output_indices: Vec<BlockOutputIndices>,
+    pub daemon_time: u64,
+    pub added_pool_txs: Vec<PoolTxInfo>,
+    pub remaining_added_pool_txids: ByteArrayVec<32>,
+}
+
+#[cfg(feature = "epee")]
+epee_object! {
+    GetBlocksResponsePoolInfoFull,
+    status: Status,
+    untrusted: bool,
+    blocks: Vec<BlockCompleteEntry>,
+    start_height: u64,
+    current_height: u64,
+    output_indices: Vec<BlockOutputIndices>,
+    daemon_time: u64,
+    added_pool_txs: Vec<PoolTxInfo>,
+    remaining_added_pool_txids: ByteArrayVec<32>,
+}
+
+#[cfg(feature = "epee")]
+/// [`EpeeObjectBuilder`] for [`GetBlocksResponse`].
+///
+/// Not for public usage.
+#[allow(dead_code, missing_docs)]
+#[derive(Clone, Debug, Default, PartialEq, Eq, PartialOrd, Ord, Hash)]
+#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
+pub struct __GetBlocksResponseEpeeBuilder {
+    pub status: Option<Status>,
+    pub untrusted: Option<bool>,
+    pub blocks: Option<Vec<BlockCompleteEntry>>,
+    pub start_height: Option<u64>,
+    pub current_height: Option<u64>,
+    pub output_indices: Option<Vec<BlockOutputIndices>>,
+    pub daemon_time: Option<u64>,
+    pub pool_info_extent: Option<PoolInfoExtent>,
+    pub added_pool_txs: Option<Vec<PoolTxInfo>>,
+    pub remaining_added_pool_txids: Option<ByteArrayVec<32>>,
+    pub removed_pool_txids: Option<ByteArrayVec<32>>,
+}
+
+#[cfg(feature = "epee")]
+impl EpeeObjectBuilder<GetBlocksResponse> for __GetBlocksResponseEpeeBuilder {
+    fn add_field<B: Buf>(&mut self, name: &str, r: &mut B) -> error::Result<bool> {
+        macro_rules! read_epee_field {
+            ($($field:ident),*) => {
+                match name {
+                    $(
+                        stringify!($field) => { self.$field = Some(read_epee_value(r)?); },
+                    )*
+                    _ => return Ok(false),
+                }
+            };
+        }
+
+        read_epee_field! {
+            status,
+            untrusted,
+            blocks,
+            start_height,
+            current_height,
+            output_indices,
+            daemon_time,
+            pool_info_extent
+        }
+
+        self.added_pool_txs = read_epee_value(r).ok().unwrap_or_default();
+        self.remaining_added_pool_txids = read_epee_value(r).ok().unwrap_or_default();
+        self.removed_pool_txids = read_epee_value(r).ok().unwrap_or_default();
```

Comment:
is this correct? these should be in `read_epee_field!` right?

Reply:
I think this was incorrect actually, since these fields are only _sometimes_ serialized (e.g. `struct GetBlocksResponsePoolInfoNone`), they won't always exist.

In that case, the fields are left as `None` where `finish()` will do the rest of the work, right?

Reply:
> they won't always exist

This should be fine in `read_epee_field` right? there is no requirement for the field the exist there.

`add_field` is called after every field name has been read, by putting these lines after `read_epee_field` we are tying to read these fields after each field has been read which isn't correct. 

> In that case, the fields are left as None where finish() will do the rest of the work, right?

This is what would happen if they were in  `read_epee_field`